File: c++/easy/find_all_numbers_disappeared_in_an_array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> miss_nums;
        vector<int> map(nums.size()+1, 0);
        for(int i = 0; i < nums.size(); ++i)
            map[nums[i]]++;
        for(int i = 1; i < map.size(); ++i)
        {
            if(map[i] == 0)
                miss_nums.push_back(i);
        }
        return miss_nums;
    }
};


Conclusion:

easy! hash_table can solve this algorithm issue
time complexity: O(n)
space complexity: O(n)

File: python/easy/find_all_numbers_disappeared_in_an_array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        map = [0 for x in range(len(nums)+1)]
        idx = 0
        ret = []
        for idx in range(0, len(nums)):
            map[nums[idx]]+=1
        idx = 0
        for idx in range(1, len(map)):
            if map[idx] == 0:
                ret.append(idx)
        return ret
